From 9b42a59736b3a9181aee80b984d7b16ca2ec1be1 Mon Sep 17 00:00:00 2001
From: Marcelo Vanzin <vanzin@cloudera.com>
Date: Thu, 13 Aug 2015 17:47:56 -0700
Subject: [PATCH 016/201] CLOUDERA-BUILD. More dependency tweaks.

Remove a few more things from the Spark assemblies, since they're
already provided in CDH.

(cherry picked from commit ec20f3dcb66ff93bed482e46f8716e7c729f852c)
---
 external/kafka/pom.xml  |    1 +
 network/shuffle/pom.xml |    1 -
 pom.xml                 |   31 +++++++++++++++++++++++--------
 sql/hive/pom.xml        |    2 +-
 4 files changed, 25 insertions(+), 10 deletions(-)

diff --git a/external/kafka/pom.xml b/external/kafka/pom.xml
index 3fe72c6..52e2182 100644
--- a/external/kafka/pom.xml
+++ b/external/kafka/pom.xml
@@ -52,6 +52,7 @@
       <groupId>org.apache.kafka</groupId>
       <artifactId>kafka_${scala.binary.version}</artifactId>
       <version>0.8.2.1</version>
+      <scope>${flume.deps.scope}</scope>
       <exclusions>
         <exclusion>
           <groupId>com.sun.jmx</groupId>
diff --git a/network/shuffle/pom.xml b/network/shuffle/pom.xml
index 00c141e..75e98550 100644
--- a/network/shuffle/pom.xml
+++ b/network/shuffle/pom.xml
@@ -46,7 +46,6 @@
     <dependency>
       <groupId>org.fusesource.leveldbjni</groupId>
       <artifactId>leveldbjni-all</artifactId>
-      <version>1.8</version>
     </dependency>
 
     <dependency>
diff --git a/pom.xml b/pom.xml
index c907755..71cbcb8 100644
--- a/pom.xml
+++ b/pom.xml
@@ -162,7 +162,7 @@
     <!-- the producer is used in tests -->
     <aws.kinesis.producer.version>0.10.1</aws.kinesis.producer.version>
     <!--  org.apache.httpcomponents/httpclient-->
-    <commons.httpclient.version>4.3.2</commons.httpclient.version>
+    <commons.httpclient.version>${cdh.httpcomponents.version}</commons.httpclient.version>
     <!--  commons-httpclient/commons-httpclient-->
     <httpclient.classic.version>3.1</httpclient.classic.version>
     <commons.math3.version>3.4.1</commons.math3.version>
@@ -188,7 +188,7 @@
     <joda.version>2.9</joda.version>
     <jodd.version>3.5.2</jodd.version>
     <jsr305.version>1.3.9</jsr305.version>
-    <libthrift.version>0.9.2</libthrift.version>
+    <libthrift.version>${cdh.thrift.version}</libthrift.version>
 
     <test.java.home>${java.home}</test.java.home>
     <test.exclude.tags></test.exclude.tags>
@@ -514,6 +514,7 @@
         <groupId>com.google.code.findbugs</groupId>
         <artifactId>jsr305</artifactId>
         <version>${jsr305.version}</version>
+        <scope>${hadoop.deps.scope}</scope>
       </dependency>
       <dependency>
         <groupId>commons-httpclient</groupId>
@@ -524,6 +525,13 @@
         <groupId>org.apache.httpcomponents</groupId>
         <artifactId>httpclient</artifactId>
         <version>${commons.httpclient.version}</version>
+        <scope>${hadoop.deps.scope}</scope>
+      </dependency>
+      <dependency>
+        <groupId>org.fusesource.leveldbjni</groupId>
+        <artifactId>leveldbjni-all</artifactId>
+        <version>1.8</version>
+        <scope>${hadoop.deps.scope}</scope>
       </dependency>
       <dependency>
         <groupId>org.apache.httpcomponents</groupId>
@@ -711,6 +719,12 @@
       </dependency>
       <dependency>
         <groupId>com.fasterxml.jackson.core</groupId>
+        <artifactId>jackson-core</artifactId>
+        <version>${fasterxml.jackson.version}</version>
+        <scope>${hadoop.deps.scope}</scope>
+      </dependency>
+      <dependency>
+        <groupId>com.fasterxml.jackson.core</groupId>
         <artifactId>jackson-databind</artifactId>
         <version>${fasterxml.jackson.version}</version>
         <scope>${hadoop.deps.scope}</scope>
@@ -880,11 +894,13 @@
         <groupId>org.apache.curator</groupId>
         <artifactId>curator-client</artifactId>
         <version>${curator.version}</version>
+        <scope>${hadoop.deps.scope}</scope>
       </dependency>
       <dependency>
         <groupId>org.apache.curator</groupId>
         <artifactId>curator-framework</artifactId>
         <version>${curator.version}</version>
+        <scope>${hadoop.deps.scope}</scope>
       </dependency>
       <dependency>
         <groupId>org.apache.curator</groupId>
@@ -1436,6 +1452,7 @@
         <groupId>${hive.group}</groupId>
         <artifactId>hive-jdbc</artifactId>
         <version>${hive.version}</version>
+        <scope>${hive.deps.scope}</scope>
         <exclusions>
           <exclusion>
             <groupId>${hive.group}</groupId>
@@ -1709,12 +1726,6 @@
         <scope>${parquet.test.deps.scope}</scope>
       </dependency>
       <dependency>
-        <groupId>com.twitter</groupId>
-        <artifactId>parquet-hadoop-bundle</artifactId>
-        <version>${hive.parquet.version}</version>
-        <scope>compile</scope>
-      </dependency>
-      <dependency>
         <groupId>org.apache.flume</groupId>
         <artifactId>flume-ng-core</artifactId>
         <version>${flume.version}</version>
@@ -1758,6 +1769,7 @@
         <groupId>org.apache.calcite</groupId>
         <artifactId>calcite-core</artifactId>
         <version>${calcite.version}</version>
+        <scope>${hive.deps.scope}</scope>
         <exclusions>
           <exclusion>
             <groupId>com.fasterxml.jackson.core</groupId>
@@ -1800,6 +1812,7 @@
         <groupId>org.apache.calcite</groupId>
         <artifactId>calcite-avatica</artifactId>
         <version>${calcite.version}</version>
+        <scope>${hive.deps.scope}</scope>
         <exclusions>
           <exclusion>
             <groupId>com.fasterxml.jackson.core</groupId>
@@ -1849,6 +1862,7 @@
         <groupId>org.apache.thrift</groupId>
         <artifactId>libthrift</artifactId>
         <version>${libthrift.version}</version>
+        <scope>${hive.deps.scope}</scope>
         <exclusions>
           <exclusion>
             <groupId>org.apache.httpcomponents</groupId>
@@ -1868,6 +1882,7 @@
         <groupId>org.apache.thrift</groupId>
         <artifactId>libfb303</artifactId>
         <version>${libthrift.version}</version>
+        <scope>${hive.deps.scope}</scope>
         <exclusions>
           <exclusion>
             <groupId>org.apache.httpcomponents</groupId>
diff --git a/sql/hive/pom.xml b/sql/hive/pom.xml
index f2cdf31..fd91b23 100644
--- a/sql/hive/pom.xml
+++ b/sql/hive/pom.xml
@@ -39,7 +39,7 @@
     <!-- Added for Hive Parquet SerDe -->
     <dependency>
       <groupId>com.twitter</groupId>
-      <artifactId>parquet-hadoop-bundle</artifactId>
+      <artifactId>parquet-hadoop</artifactId>
     </dependency>
     <dependency>
       <groupId>org.apache.spark</groupId>
-- 
1.7.9.5

